libecole/include/ecole/environment/environment.hpp

```diff
@@ -109,10 +116,18 @@ class Environment {
 			reward_function().before_reset(model());
 			information_function().before_reset(model());
 			auto const [done, action_set] = dynamics().reset_dynamics(model(), std::forward<Args>(args)...);
-
 			can_transition = !done;
+
+			OptionalObservation observation;
+			if (can_transition) {
```

Comment:
I would suggest `if (not done)` to keep the C++ code more in-sync with the python code.